diff --git a/src/cmd/link/internal/ld/ldelf.go b/src/cmd/link/internal/ld/ldelf.go
index d4f9fc4..87dabaa 100644
--- a/src/cmd/link/internal/ld/ldelf.go
+++ b/src/cmd/link/internal/ld/ldelf.go
@@ -799,6 +799,13 @@ func ldelf(ctxt *Link, f *bio.Reader, pkg string, length int64, pn string) {
 				continue
 			}
 
+			if sect.name == ".debug_str" && sym.name == "L0" && sym.type_ == 0 {
+				// introduced by https://sourceware.org/git/?p=binutils-gdb.git;a=commit;h=49fced1206db40c71208c201165d65f92c69cebe on s390x
+				// TODO investigate from where they are actually coming from, could be possible issue with elf parsing as seeing 0x1 in name is weird
+				// See issue https://github.com/golang/go/issues/20996
+				continue
+			}
+
 			if strings.HasPrefix(sym.name, ".LASF") { // gcc on s390x does this
 				continue
 			}
